dbms/src/Flash/Coprocessor/tests/gtest_streaming_writer.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Flash/Coprocessor/ArrowChunkCodec.h>
+#include <Flash/Coprocessor/CHBlockChunkCodec.h>
+#include <Flash/Coprocessor/DefaultChunkCodec.h>
+#include <Interpreters/Context.h>
+#include <Storages/Transaction/TiDB.h>
+#include <TestUtils/TiFlashTestBasic.h>
+#include <TestUtils/TiFlashTestEnv.h>
+#include <gtest/gtest.h>
+
+#include <Flash/Coprocessor/StreamingDAGResponseWriter.cpp>
+#include <iostream>
+
+namespace DB
+{
+namespace tests
+{
+class TestStreamingWriter : public testing::Test
+{
+protected:
+    void SetUp() override
+    {
+        dag_context_ptr = std::make_unique<DAGContext>(1024);
+        dag_context_ptr->is_mpp_task = true;
+        dag_context_ptr->is_root_mpp_task = true;
+        dag_context_ptr->result_field_types = makeFields();
+        context.setDAGContext(dag_context_ptr.get());
+    }
+
+public:
+    TestStreamingWriter()
+        : context(TiFlashTestEnv::getContext())
+    {}
+
+    // Return 10 Int64 column.
+    static std::vector<tipb::FieldType> makeFields()
+    {
+        std::vector<tipb::FieldType> fields(10);
+        for (int i = 0; i < 10; ++i)
+        {
+            fields[i].set_tp(TiDB::TypeLongLong);
+        }
+        return fields;
+    }
+
+    static DAGSchema makeSchema()
+    {
+        auto fields = makeFields();
+        DAGSchema schema;
+        for (size_t i = 0; i < fields.size(); ++i)
+        {
+            ColumnInfo info = TiDB::fieldTypeToColumnInfo(fields[i]);
+            schema.emplace_back(String("col") + std::to_string(i), std::move(info));
+        }
+        return schema;
+    }
+
+    // Return a block with **rows** and 10 Int64 column.
+    static Block prepareBlock(const std::vector<Int64> & rows)
```

Comment:
Can use ColumnGenerator in future, If we support uniform data gen

Reply:
`ColumnGenerator` used  in `TestMPPExchangeWriter::prepareRandomBlock`